src/video/wayland/SDL_waylanddatamanager.c

```diff
@@ -517,6 +517,7 @@ SDL_bool Wayland_primary_selection_offer_has_mime(SDL_WaylandPrimarySelectionOff
 void Wayland_data_offer_destroy(SDL_WaylandDataOffer *offer)
 {
     if (offer != NULL) {
+        wl_data_offer_finish(offer->offer);
```

Comment:
This should only be called upon successful completion of a drag & drop operation. Calling it every time an offer is destroyed isn't correct and I occasionally see errors from it when running the drag & drop test.

Also, as this function is only available in version 3 of the data manager or higher, a version check is needed before calling it to make sure the data manager version is >= `WL_DATA_OFFER_FINISH_SINCE_VERSION`

Reply:
Should I add a function `Wayland_data_offer_finish` and call it after a successfully handled drop?

Reply:
I was going to say that a helper function for this little snippet only needed in one place seems to be unnecessary, but you went that way anyways, so it's irrelevant :)